librz/core/cbin.c

```diff
@@ -4481,3 +4481,82 @@ RZ_API char *rz_core_bin_method_flags_str(ut64 flags, int mode) {
 out:
 	return rz_strbuf_drain(buf);
 }
+
+static RzCoreStringType get_string_type(const ut8 *buf, ut64 len) {
+	rz_return_val_if_fail(buf && len, RZ_CORE_STRING_KIND_UNKNOWN);
+	ut64 needle = 0;
+	int rc, i;
+	RzCoreStringType str_type = RZ_CORE_STRING_KIND_UNKNOWN;
+
+	while (needle < len) {
+		rc = rz_utf8_decode(buf + needle, len - needle, NULL);
+		if (!rc) {
+			needle++;
+			continue;
+		}
+		if (needle + rc + 2 < len &&
+			buf[needle + rc + 0] == 0x00 &&
+			buf[needle + rc + 1] == 0x00 &&
+			buf[needle + rc + 2] == 0x00) {
+			str_type = RZ_CORE_STRING_KIND_WIDE16;
+		} else {
+			str_type = RZ_CORE_STRING_KIND_ASCII;
+		}
+		for (i = 0; needle < len; i += rc) {
+			RzRune r;
+			if (str_type == RZ_CORE_STRING_KIND_WIDE16) {
+				if (needle + 1 < len) {
+					r = buf[needle + 1] << 8 | buf[needle];
+					rc = 2;
+				} else {
+					break;
+				}
+			} else {
+				rc = rz_utf8_decode(buf + needle, len - needle, &r);
+				if (rc > 1) {
+					str_type = RZ_CORE_STRING_KIND_UTF8;
+				}
+			}
+			/*Invalid sequence detected*/
+			if (!rc) {
+				needle++;
+				break;
+			}
+			needle += rc;
+		}
+	}
+	return str_type;
+}
+
+/**
+ * \brief Returns the information about string given the offset and legnth
+ *
+ * \param core RzCore instance
+ * \param block Pointer to the string
+ * \param len Length of the string
+ * \param type A type of the string to override autodetection
+ */
+RZ_OWN RZ_API RzCoreString *rz_core_string_information(RzCore *core, const char *block, int len, RzCoreStringType type) {
```

Comment:
mmm if you already know the offset of the string and the length what kind of additional information is provided in `RzCoreString`?

Reply:
Mostly type and the section name that contains it.

Reply:
Also note, that the length in question is different from size, we have to provide/store both.

Reply:
There exist a `RzBinString` which has a similar structure but slightly different. I am waiting for @thestr4ng3r opinion, maybe we can change RzBinString instead and reuse it.

Reply:
I would expect structures in RzCore to be a mix of things coming from different modules (e.g. RzBin, RzIO, etc.). If all the information are actually RzBin specific, maybe it's just better to keep the "string" concept in RzBin. Though honestly I'm not that familiar with this thing.

Reply:
It is also loosely related to https://github.com/rizinorg/rizin/issues/38#issuecomment-763535238 but for another time.
In short - we need to unify string detection/analysis/display between all these places.